InvokerSupervisionTests heisenbug

Following failure was seen on travis

```
whisk.core.loadBalancer.test.InvokerSupervisionTests » InvokerPool should forward an ActivationMessage to the sendActivation-Method

Unsatisfied expectation: Expected: inAnyOrder { StubFunction2-3(whisk.system/utils/date?message={}, InstanceId(0,None)) any number of times (never called) StubFunction2-3(whisk.system/utils/date?message={}, InstanceId(0,None)) once (never called - UNSATISFIED) } Actual:

[2018-06-25T10:55:08.596Z] [INFO] [#tid_sid_invokerHealth] [InvokerPool] invoker status changed to 0 -> Healthy
[2018-06-25T10:55:08.611Z] [INFO] [#tid_sid_dispatcher] [MessageFeed] handler capacity = 4, pipeline fill at = 4, pipeline depth = 8
[2018-06-25T10:55:08.641Z] [ERROR] Can't log call to mock object, have expectations been verified already?
java.lang.RuntimeException: Can't log call to mock object, have expectations been verified already?
	at org.scalamock.function.FakeFunction.handle(FakeFunction.scala:46)
	at org.scalamock.function.FakeFunction2.apply(FakeFunction.scala:68)
	at whisk.core.loadBalancer.InvokerPool$$anonfun$receive$1.applyOrElse(InvokerSupervision.scala:116)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:517)
	at whisk.core.loadBalancer.InvokerPool.aroundReceive(InvokerSupervision.scala:73)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:588)
	at akka.actor.ActorCell.invoke(ActorCell.scala:557)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
[2018-06-25T10:55:08.653Z] [INFO] [#tid_sid_dispatcher] [MessageFeed] handler capacity = 4, pipeline fill at = 4, pipeline depth = 8

```

https://scans.gradle.com/s/l23drmohxtvam/tests/xaqugzlgs2zw6-pe5egvx7auofs

--- a/tests/src/test/scala/common/LoggedFunction.scala
+++ b/tests/src/test/scala/common/LoggedFunction.scala
@@ -83,10 +83,11 @@ class LoggedFunction6[A1, A2, A3, A4, A5, A6, B](body: (A1, A2, A3, A4, A5, A6) 
 }
 
 object LoggedFunction {
-  def apply[A1, B](body: (A1) => B) = new LoggedFunction1(body)
-  def apply[A1, A2, B](body: (A1, A2) => B) = new LoggedFunction2(body)
-  def apply[A1, A2, A3, B](body: (A1, A2, A3) => B) = new LoggedFunction3(body)
-  def apply[A1, A2, A3, A4, B](body: (A1, A2, A3, A4) => B) = new LoggedFunction4(body)
-  def apply[A1, A2, A3, A4, A5, B](body: (A1, A2, A3, A4, A5) => B) = new LoggedFunction5(body)
-  def apply[A1, A2, A3, A4, A5, A6, B](body: (A1, A2, A3, A4, A5, A6) => B) = new LoggedFunction6(body)
+  def apply[A1, B](body: (A1) => B) = new LoggedFunction1[A1, B](body)
+  def apply[A1, A2, B](body: (A1, A2) => B) = new LoggedFunction2[A1, A2, B](body)
+  def apply[A1, A2, A3, B](body: (A1, A2, A3) => B) = new LoggedFunction3[A1, A2, A3, B](body)
+  def apply[A1, A2, A3, A4, B](body: (A1, A2, A3, A4) => B) = new LoggedFunction4[A1, A2, A3, A4, B](body)
+  def apply[A1, A2, A3, A4, A5, B](body: (A1, A2, A3, A4, A5) => B) = new LoggedFunction5[A1, A2, A3, A4, A5, B](body)
+  def apply[A1, A2, A3, A4, A5, A6, B](body: (A1, A2, A3, A4, A5, A6) => B) =
+    new LoggedFunction6[A1, A2, A3, A4, A5, A6, B](body)
 }

--- a/tests/src/test/scala/whisk/core/loadBalancer/test/InvokerSupervisionTests.scala
+++ b/tests/src/test/scala/whisk/core/loadBalancer/test/InvokerSupervisionTests.scala
@@ -42,7 +42,7 @@ import akka.testkit.TestFSMRef
 import akka.testkit.TestKit
 import akka.testkit.TestProbe
 import akka.util.Timeout
-import common.StreamLogging
+import common.{LoggedFunction, StreamLogging}
 import whisk.common.TransactionId
 import whisk.core.WhiskConfig
 import whisk.core.connector.ActivationMessage
@@ -175,7 +175,9 @@ class InvokerSupervisionTests
     val invokerName = s"invoker${invokerInstance.toInt}"
     val childFactory = (f: ActorRefFactory, instance: InvokerInstanceId) => invoker.ref
 
-    val sendActivationToInvoker = stubFunction[ActivationMessage, InvokerInstanceId, Future[RecordMetadata]]
+    val sendActivationToInvoker = LoggedFunction { (a: ActivationMessage, b: InvokerInstanceId) =>
+      Future.successful(new RecordMetadata(new TopicPartition(invokerName, 0), 0L, 0L, 0L, Long.box(0L), 0, 0))
+    }
 
     val supervisor = system.actorOf(InvokerPool.props(childFactory, sendActivationToInvoker, pC))
 
@@ -196,18 +198,10 @@ class InvokerSupervisionTests
       content = None)
     val msg = ActivationRequest(activationMessage, invokerInstance)
 
-    sendActivationToInvoker
-      .when(activationMessage, invokerInstance)
-      .returns(
-        Future.successful(new RecordMetadata(new TopicPartition(invokerName, 0), 0L, 0L, 0L, Long.box(0L), 0, 0)))
-
     supervisor ! msg
 
     // Verify, that MessageProducer will receive a call to send the message
-    retry(
-      sendActivationToInvoker.verify(activationMessage, invokerInstance).once,
-      N = 3,
-      waitBeforeRetry = Some(500.milliseconds))
+    retry(sendActivationToInvoker.calls should have size 1, N = 3, waitBeforeRetry = Some(500.milliseconds))
   }
 
   behavior of "InvokerActor"

